### PR TITLE
Refactoring cache usage and cache look up for convergence

### PR DESCRIPTION
**Pull request by omercs**
Refactoring step1
UserIdentifier
TokenCache
<a href='#crh-start'></a><a href='#crh-data-%7B%22approved%22%3A%20%7B%22https%3A//github.com/RPangrle-MS%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7004783%3Fv%3D3%22%7D%2C%20%22https%3A//github.com/emily54321%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12581508%3Fv%3D3%22%7D%7D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/392%23issuecomment-117340214%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222015-06-30T20%3A58%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12581508%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/emily54321%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222015-06-30T22%3A39%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7004783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RPangrle-MS%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20f712342327336cce17f1602a33c2f90ae340e2d2%20samples/hello/src/com/microsoft/aad/adal/hello/MainActivity.java%20116%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/392%23discussion_r33613360%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22On%20ADAL.NET%20the%20unique%20ID%20is%20some%20ugly%20non-human%20readable%20string%20%28the%20subject%20or%20object%20id%29.%20Does%20that%20map%20to%20the%20previous%20functionality%20of%20getUserId%28%29%20here%3F%20I%20would%20think%20displayable%20id%20would%20be%20closer%20to%20what%20you%20want.%22%2C%20%22created_at%22%3A%20%222015-06-30T19%3A27%3A26Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7004783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RPangrle-MS%22%7D%7D%2C%20%7B%22body%22%3A%20%22Right%2C%20this%20was%20in%20sample%20app.%20I%20will%20update%20that.%22%2C%20%22created_at%22%3A%20%222015-06-30T20%3A56%3A36Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/466805%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/omercs%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20samples/hello/src/com/microsoft/aad/adal/hello/MainActivity.java%3AL260-277%22%7D%7D%2C%20%22processed%22%3A%20%5B%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/392%23issuecomment-117340214%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/392%23issuecomment-117363850%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/392%23discussion_r33613360%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/392%23discussion_r33622756%22%5D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/RPangrle-MS'></a><a href='https://github.com/emily54321'></a>

- [ ] <a href='#crh-comment-Pull f712342327336cce17f1602a33c2f90ae340e2d2 samples/hello/src/com/microsoft/aad/adal/hello/MainActivity.java 116'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/392#discussion_r33613360'>File: samples/hello/src/com/microsoft/aad/adal/hello/MainActivity.java:L260-277</a></b>
- <a href='https://github.com/RPangrle-MS'></a> On ADAL.NET the unique ID is some ugly non-human readable string (the subject or object id). Does that map to the previous functionality of getUserId() here? I would think displayable id would be closer to what you want.
- <a href='https://github.com/omercs'></a> Right, this was in sample app. I will update that.

<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-android/pull/392?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-android/pull/392?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-android/pull/392?approve=0'></a>&nbsp;<a href='https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/392'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by RandalliLama** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/392'>Refresh</a>.<a href='#crh-update-none'></a>

File: samples/hello/src/com/microsoft/aad/adal/hello/MainActivity.java [hunk 1]
```diff
@@ -302,15 +260,17 @@ private void displayMessage(String msg) {
 
     private String getUserId() {
         if (mResult != null && mResult.getUserInfo() != null
-                && mResult.getUserInfo().getUserId() != null) {
-            return mResult.getUserInfo().getUserId();
+                && mResult.getUserInfo().getUniqueId() != null) {
+            return mResult.getUserInfo().getUniqueId();
```
Comment (RPangrle-MS) [inline samples/hello/src/com/microsoft/aad/adal/hello/MainActivity.java]: On ADAL.NET the unique ID is some ugly non-human readable string (the subject or object id). Does that map to the previous functionality of getUserId() here? I would think displayable id would be closer to what you want.

Comment (omercs) [inline samples/hello/src/com/microsoft/aad/adal/hello/MainActivity.java]: Right, this was in sample app. I will update that.

**Comment by emily54321** [comment]
:shipit:

**Comment by RPangrle-MS** [comment]
:shipit:

**Pull request merged**